okhttp/src/main/java/okhttp3/internal/ws/WebSocketOptions.kt

```diff
@@ -0,0 +1,56 @@
+package okhttp3.internal.ws
+
+import okhttp3.Response
+import java.io.IOException
+
+data class WebSocketOptions(
+    @JvmField val compressionEnabled: Boolean,
+    @JvmField val contextTakeover: Boolean
+) {
+  companion object {
+    private const val OPTION_CLIENT_NO_CONTEXT_TAKEOVER = "client_no_context_takeover"
+    private const val OPTION_CLIENT_MAX_WINDOW_BITS = "client_max_window_bits"
+    private const val SUPPORTED_CLIENT_MAX_WINDOW_BITS = 15
+
+    private val DEFAULT = WebSocketOptions(
+        compressionEnabled = false,
+        contextTakeover = false
+    )
+
+    @Throws(IOException::class)
+    @JvmStatic
+    fun parseServerResponse(response: Response): WebSocketOptions =
+        response.header("Sec-WebSocket-Extensions")
+            ?.takeIf { it.contains("permessage-deflate") }
```

Comment:
I’m tempted to ask you to use less Kotlin and more if/else

Reply:
Sure. Any details on why, so that I can better follow in the future?

Reply:
I find this method in particular a little Perl-ish. I need to study this to learn what it’s really doing. Maybe some day I’ll be good enough at Kotlin where I can read this naturally, but for now I think this code loses readability to be compact.

Reply:
OK, no problem.
A bigger rewrite of the WebSocketOptions parsing is on its way, I've streamlined the code there.